File: SCC0600/exerciciosC/ex34.c
#include<stdio.h>

int main(void) {
    int input;
    scanf("%d", &input);
    char res = 'N';

    int somadedivisores = 0;
    for (int i = 1; i < input; i++) {
        if (input % i == 0) {
            somadedivisores += i;
        }
    }

    if (input == somadedivisores) {
        res = 'S';
    }

    printf("%c", res);
}

File: SCC0600/exercíciosPy/ex34.py
n = int(input())

res = 'N'
i = 1
algoritmo = n-1

if n > 1:
  while algoritmo <= n:
    algoritmo = (2**(i-1) * (2**i - 1))
    if algoritmo == n:
      res = 'S'
      break
    i += 1

print(res)